vega/base.py

```diff
@@ -25,31 +24,24 @@ def __init__(self, spec, data=None, opt=None):
     def _prepare_spec(self, spec, data):
         return spec
 
-    def _generate_html(self, id):
-        template = utils.get_content(self.HTML_TEMPLATE)
-        return template.format(id=id)
-
     def _generate_js(self, id, **kwds):
         template = utils.get_content(self.JS_TEMPLATE)
         selector = '#{0}'.format(id)
         payload = template.format(
             selector=selector,
+            id=json.dumps(str(id)),
             spec=json.dumps(self.spec, **kwds),
             opt=json.dumps(self.opt, **kwds),
             type=self.render_type
         )
         return payload
 
-    def _ipython_display_(self):
+    def _repr_mimebundle_(self, include=None, exclude=None):
```

Comment:
Definitely a fan here, thanks for doing this!

Reply:
Does the classic notebook support ``_repr_mimebundle_`` now?

Reply:
The notebook itself has always supported mime bundles (all the way back to the early days of the notebook). It's the IPython / ipykernel side that has to support this standard in python. It has since last year (?). Matthias or Thomas Kluyver would know best.

Reply:
Ah, right. I was mixing up issues in my mind. It's the vega/vega-lite specific mimebundle that doesn't work in the notebook, I think.

Reply:
Yep! Notebook has to use the js or html payload, at least until someone provides vega support directly to classic. I personally think we should enable it by default and not make it an extension.